app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -18,14 +19,34 @@ class CiYaml {
   });
 
   /// The underlying protobuf that contains the raw data from .ci.yaml.
-  final pb.SchedulerConfig config;
+  pb.SchedulerConfig config;
 
   /// The [RepositorySlug] that [config] is from.
   final RepositorySlug slug;
 
   /// The git branch currently being scheduled against.
   final String branch;
 
+  /// Load [currentConfigYaml] (and optionally [totConfigYaml] ) to [pb.SchedulerConfig] and validate the dependency graph.
+  /// where totYaml means tip of tree Yaml.
+  CiYaml.fromYaml(YamlMap? currentConfigYaml, {YamlMap? totConfigYaml})
+      : branch = '',
+        slug = RepositorySlug('flutter', 'fake'),
```

Comment:
Don't make these fake values. These should be required (following something similar as the existing constructor)